src/lib/data/storage/PersistentStorage.cpp

```diff
@@ -1024,35 +1024,45 @@ std::shared_ptr<Graph> PersistentStorage::getGraphForAll() const
 {
 	TRACE();
 
-	std::vector<Id> tokenIds;
-
-	m_sqliteIndexStorage.forEach<StorageNode>([&](StorageNode&& node) {
-		bool showNode = true;
-		if (m_symbolDefinitionKinds.size())
-		{
-			auto it = m_symbolDefinitionKinds.find(node.id);
-			showNode = (it != m_symbolDefinitionKinds.end() && it->second == DEFINITION_EXPLICIT);
-		}
-
-		if (showNode &&
-			(NodeType(NodeType::intToType(node.type)).isPackage() ||
-			 !m_hierarchyCache.isChildOfVisibleNodeOrInvisible(node.id)))
-		{
-			tokenIds.push_back(node.id);
-		}
-	});
-
-	for (const auto& p: m_fileNodeIndexed)
+	std::shared_ptr<Graph> graph = std::make_shared<Graph>();
+	if (m_symbolDefinitionKinds.size()) {
+		m_sqliteIndexStorage.forEach<StorageNode>([&](StorageNode&& storageNode) {
+			const NodeType type(NodeType::intToType(storageNode.type));
+			if (type.isFile())
+			{
+				addFileNodeToGraph(storageNode, graph.get());
+			}
+			else
+			{
+				auto sdk_it = m_symbolDefinitionKinds.find(storageNode.id);
+				bool showNode = (sdk_it != m_symbolDefinitionKinds.end() && sdk_it->second == DEFINITION_EXPLICIT);
+				if (showNode && (
+					type.isPackage() ||
+					!m_hierarchyCache.isChildOfVisibleNodeOrInvisible(storageNode.id)
+					)
+				)
+				{
+					NameHierarchy nameHierarchy = NameHierarchy::deserialize(storageNode.serializedName);
+					graph->createNode(storageNode.id, type, std::move(nameHierarchy), DEFINITION_EXPLICIT);
```

Comment:
Put this in a reusable `addNodeToGraph` method as described [here](https://github.com/CoatiSoftware/Sourcetrail/pull/890#issuecomment-573931961). The way nodes are added to the graph should not be duplicated.

In case where there are no `m_symbolDefinitionKinds` the overview shows all symbols. You can't just always set `DEFINITION_EXPLICIT`, you need the lookup.